pkg/ibmvpc/ibmDrawioGenerator.go

```diff
@@ -5,6 +5,17 @@ import (
 	"github.com/np-guard/vpc-network-config-analyzer/pkg/vpcmodel"
 )
 
+func (v *VPC) IsExternal() bool                  { return false }
+func (z *Zone) IsExternal() bool                 { return false }
+func (s *Subnet) IsExternal() bool               { return false }
+func (sgl *SecurityGroupLayer) IsExternal() bool { return false }
+func (nl *NaclLayer) IsExternal() bool           { return false }
+func (n *IKSNode) IsExternal() bool              { return false }
+func (ni *NetworkInterface) IsExternal() bool    { return false }
+func (v *Vsi) IsExternal() bool                  { return false }
+func (pgw *PublicGateway) IsExternal() bool      { return false }
+func (fip *FloatingIP) IsExternal() bool         { return false }
```

Comment:
An alternative is to add the method `IsExternal() bool` to the `VPCResource` struct in `vpcmodel/abstractVPC.go` such that it just returns `true` there , and also change the type `ExternalNetwork` in `vpcmodel/externalNetwork.go` so that it will no longer contain `VPCResource` (and add its methods to implement the `VPCResourceIntf` ).

Reply:
done